### ISSUE TITLE
Ready for production use?

### ISSUE THREAD

**Issue by kbrandwijk** [samcrow/rust-xplm#10]
I came across this great repo wanting to start writing X-Plane plugins in Rust instead of C++. I've read the readme, and the notes about drawing and display SDK parts needing 'some more work'. Is it realistic to start down the Rust path? Or will I find myself in a dead end because of missing SDK wrappers? I don't have much Rust knowledge, so I don't know how easy or hard it would be to contribute any missing pieces down the road.

What would be your advice at this point in the development?

**Comment by samcrow** [comment]
If you want to make a plugin that works with datarefs and commands, using Rust and this library should work without too much trouble.

Graphics and GUI are the main areas where Rust and the plugin APIs don't easily work together, and the Rust bindings are incomplete/missing. For those, C++ and C are better options.

Because the documentation for this library is currently not very detailed, it's good to have some experience with X-Plane plugins in C or C++ before using Rust.

**Comment by Ev1dentSnow** [comment]
> If you want to make a plugin that works with datarefs and commands, using Rust and this library should work without too much trouble.
> 
> Graphics and GUI are the main areas where Rust and the plugin APIs don't easily work together, and the Rust bindings are incomplete/missing. For those, C++ and C are better options.
> 
> Because the documentation for this library is currently not very detailed, it's good to have some experience with X-Plane plugins in C or C++ before using Rust.

Would you recommend writing plugin code in Rust and then the GUI portion of a plugin in C++ using something like this? https://github.com/ocornut/imgui

**Comment by samcrow** [comment]
Yes, I haven't done that myself but it might be a good approach. If you can get that GUI library to work in the X-Plane environment. it should be fairly easy to use.
